### ISSUE TITLE
access to personal drafts collection denied ("unknown error")

### ISSUE THREAD

**Issue by tinkerbell1704** [joostdecock/site#647]
Dear Freesewing Team,

I am extremely happy to be part of the community and be able to finally sew a tailoerd garment to my husband, which completely fits him. I have tried a hat and the Bruce pants so far.

However, I have found something bad since yesterday, namely I do not have access to my personal drafts collection any longer, it says "an unknown error occured, please report it github.com". 

I do not know exactly why this error message came out of the blue, It would be nice to have them in the place just for easy access. I know it is not a big deal to re-create a draft again from the model's data stored, but still, it would be much better to have them together there.

Please fix the problem.
Many thanks for your help in advance,
tinkerbell1704

**Comment by joostdecock** [comment]
Hi @tinkerbell1704 thanks for reporting this.
Would it be possible to share the url, or the darft reference, so we can try to reproduce this issue?

**Comment by tinkerbell1704** [comment]
Hi Joost,
Thanks for your prompt answer. Sure, these are the ones I created yesterday:
freesewing.org/drafts/mkrzp
freesewing.org/drafts/cwxuk

However, just realised if I put this in the address line of my browser, I can jump directly to the drafts, which means the drafts are still exist on the page but not visible from my draft collection. Indeed, to get access to them I have to remember the odd draft codes - which is not easy, unless I have saved on the desktop or printed out the draft already. 

I have a print screen picture as well to show what I can see by clicking on the personal draft collection under my name, but I cannot attach a file here - or I have not found it yet how to.

Many thanks for your answer,
tinkerbell1704

**Comment by tinkerbell1704** [comment]
Actually, I figured out how to attach a screenshot, this is what I am faced with by clicking personal draft collection button:
![screenshot_28_02_2019](https://user-images.githubusercontent.com/48083575/53554925-27e8ec00-3b41-11e9-9c29-934f1dbd4454.png)

**Comment by joostdecock** [comment]
Thank you, I was able to reproduce the issue. There's an error on that page:

```
t.$store.state.user.models[e.item.model] is undefined
```

It seems to be something related to one of your models, but I'd need to look deeper into it.
I'll try to get it fixed tonight, as I'm not at liberty to work on it right now.

In the meanwhile, here are links to all your drafts:

 - https://freesewing.org/drafts/nmcqz
 - https://freesewing.org/drafts/ehmga
 - https://freesewing.org/drafts/anbhe
 - https://freesewing.org/drafts/gxhnm
 - https://freesewing.org/drafts/dvhxe
 - https://freesewing.org/drafts/auzmt
 - https://freesewing.org/drafts/mgkfw
 - https://freesewing.org/drafts/mkrzp
 - https://freesewing.org/drafts/cwxuk

**Comment by tinkerbell1704** [comment]
Many thanks, Joost, for your effort to reproduce the issue and sending me the urls to my darfts.  

Yes, these are the ones I made earlier, however I  already deleted some of them (the first 5), as they were just tests in the beginning to be able to familiarize with the system. I would keep only the last 4 anyway. 

But do not bother to fix the drafts, I can remake them very easy, they are not complicated at all. My only points are to get an access again to my draft folder for the future, and to know what wrong I have done unintentionally leading to this issue to be able to avoid it in the future. 

Many-many thanks for your help.

**Comment by joostdecock** [comment]
This was due to the fact that you deleted some models, but still had drafts that used those models.
There was an unhandled exception when we tried to display the model name in your draft table, as the model no longer exists.

I've implemented a fix, and deployed it in production. I've tested it, and you can see your draft list now.

Thanks again for reporting this.

**Comment by tinkerbell1704** [comment]
I am extremely grateful for your immediate help, Joost, everything is fine now, I could check it this morning. 

That is true, I 've deleted some models as it was not clear for me in the beginning how the system works, and I tried different options.to be familiarised with it. Of course, I did not know that deleting a modell will cause a page error. 

Thank you once again, you are fantastic, I love your page and your attitude and your enthusiasm how you run it. Have a nice day :)
